apps/testingaccessibility/src/pages/welcome/index.tsx

```diff
@@ -9,111 +9,65 @@ import Layout from 'components/app/layout'
 import Image from 'next/image'
 import prisma from '../../db'
 import Link from 'next/link'
+import {stripeData} from '../../utils/record-new-purchase'
+import {getPurchaseDetails} from '../../lib/purchases'
+import {isString} from 'lodash'
 
 export const getServerSideProps: GetServerSideProps = async ({req, query}) => {
-  const {purchaseId} = query
+  const {purchaseId: purchaseQueryParam, session_id, upgrade} = query
   const token = await getToken({
     req,
     secret: process.env.NEXTAUTH_SECRET,
   })
 
-  if (purchaseId) {
-    const allPurchases = await prisma.purchase.findMany({
-      where: {
-        userId: token?.sub,
-      },
-      select: {
-        id: true,
-        productId: true,
-      },
-    })
+  let purchaseId = purchaseQueryParam
+
+  if (session_id) {
+    const {stripeChargeId} = await stripeData(session_id as string)
     const purchase = await prisma.purchase.findFirst({
       where: {
-        id: purchaseId as string,
-        userId: token?.sub,
-      },
-      select: {
-        merchantChargeId: true,
-        bulkCoupon: {
-          select: {
-            id: true,
-            maxUses: true,
-            usedCount: true,
-          },
-        },
-        product: {
-          select: {
-            id: true,
-            name: true,
-          },
+        merchantCharge: {
+          identifier: stripeChargeId,
         },
       },
     })
 
-    if (!purchase) {
+    if (purchase) {
+      purchaseId = purchase.id
+    } else {
       return {
         redirect: {
-          destination: '/learn',
+          destination: `/thanks/purchase?session_id=${session_id}`,
```

Comment:
here we have a session id, but couldn't find a purchase so we will bounce them over to /thanks and tell them to check inbox